fastlane/Fastfile

```diff
@@ -447,28 +447,53 @@ platform :ios do
   lane :generate_docs do
     ENV["INCLUDE_DOCC_PLUGIN"] = "true"
     version_number = current_version_number
-    docs_folder = "docs/#{version_number}"
-    hosting_base_path = "purchases-ios/#{version_number}"
-
-    Dir.chdir("..") do
-      sh("swift",
-         "package",
-         "--disable-sandbox",
-         "--allow-writing-to-directory",
-         docs_folder,
-         "generate-documentation",
-         "--target",
-         "RevenueCat",
-         "--disable-indexing",
-         "--output-path",
-         docs_folder,
-         "--hosting-base-path",
-         "#{hosting_base_path}",
-         "--transform-for-static-hosting")
+    hosting_base_path = "purchases-ios-docs/#{version_number}"
+    docs_repo_base_url = "https://github.com/revenuecat/"
+    docs_repo_name = "purchases-ios-docs"
+    docs_repo_url = docs_repo_base_url + docs_repo_name
+    
+    Dir.mktmpdir do |docs_generation_folder|
+      # swift package must be run from the dir that contains the Package.swift
+      # output is generated in docs_generation_folder
+      Dir.chdir("..") do
+        sh("swift",
+           "package",
+           "--disable-sandbox",
+           "--allow-writing-to-directory",
+           docs_generation_folder,
+           "generate-documentation",
+           "--target",
+           "RevenueCat",
+           "--disable-indexing",
+           "--output-path",
+           docs_generation_folder,
+           "--hosting-base-path",
+           "#{hosting_base_path}",
+           "--transform-for-static-hosting")
+        docs_index_path = "#{Dir.pwd}/scripts/docs/index.html"
+        # clone docs repo
+        Dir.mktmpdir do |docs_repo_clone_dir|
+          Dir.chdir(docs_repo_clone_dir) do
+            sh("git", "clone", docs_repo_url)
+            Dir.chdir(docs_repo_name) do
+              # copy docs generated in the previous step into the docs folder
+              # and push the changes
+              docs_destination_folder = "docs/#{version_number}"
+              index_destination_path = "docs/index.html"
+              FileUtils.mv docs_generation_folder, docs_destination_folder
+              FileUtils.cp docs_index_path, index_destination_path
+
+              # using sh instead of fastlane commands because fastlane would run
+              # from the repo root
+              sh("git", "add", docs_destination_folder)
+              sh("git", "add", index_destination_path)
+              sh("git", "commit", "-m", 'Update documentation')
```

Comment:
Maybe put the version number in the commit?

```suggestion
              sh("git", "commit", "-m", "Update documentation for #{verison_number}")
```

Reply:
great call!